### DISCUSSION TITLE
YARP video viewer issues

### DISCUSSION THREAD

**Discussion by patrickshas** [robotology/community#675] [Q&A] [answered]
Hi all,
So we are displaying a processed video feed "icub/cameracalib/right/out" using a viewer port. The python process which creates this video feed takes in the undistorted right eye video feed and does some processing, adding a bounding box and some vectors to the video feed before it is displayed. As it could be seen in the video attached here (once the computational load gets heavy/ we are assuming this is what's happening) the displayed video starts switching between distorted and undistorted.
Any ideas what is happening here?
Thanks in advance.

https://github.com/robotology/community/assets/45318225/07c49e4d-aa71-4e2f-9168-ed35907fce3b

**Comment by pattacini** [comment]
Hi @patrickshas 

Would you be willing to repeat your experiment keeping an eye on the uncalibrated images too?
You can connect the two viewers above on the left to the uncalibrated streams provided via the ports `/icub/cam/left` and `/icub/cam/right`.

This way, we should be able to confirm whether `camCalib` is playing a role here.

**Comment by patrickshas** [comment]
Hi @pattacini,
I am attaching the requested video here.
Seems like the issue is not present when using the uncalibrated stream.

https://github.com/robotology/community/assets/45318225/50acec55-d4d9-4950-80c5-98e4e07665c2

**Reply by pattacini** [reply]
Hi @patrickshas 

Another test to perform for us.
Could you please run a `yarp ping` for all the ports involved here and report the output?

For example:
```console
yarp ping /icub/cam/left
```

Let's ping all the ports:
- `/icub/cam/[left|right]`
- `/icub/camcalib/[left|right]/[in|out]`
- `/icub/view/[left|right]`
- your module's input ports

Ideally, we should do that for both conditions: when you use calibrated and uncalibrated images. 

Thanks!

**Comment by patrickshas** [comment]
Hello @pattacini,
Here are the results for that test.
Apologies for the delay,
Thanks again

[ping test.txt](https://github.com/robotology/community/files/15253701/ping.test.txt)

**Comment by pattacini** [comment, marked as answer]
Hi @patrickshas 

From the log, everything seems to be in order from the standpoint of cam/camcalib YARP port connections.

For doubt's sake, could you `yarp ping` the input ports opened by your module, in both cases, with calibrated and with uncalibrated images?
- `yarp ping /ptgaze-in-left`
- `yarp ping /ptgaze-in-right`

Also, a few other steps that would help debug the problem:
- I assume that the system was working as expected some time ago and then started misbehaving. What happened? Any SW update?
- What is the version of `icub-main` you're building? Git tag or Git hash are ok.
- You may consider dumping the calibrated images streamed out by camCalib via the tool [yarpdatadumper](https://yarp.it/latest/group__yarpdatadumper.html). Then, you could zip them and post the archive here.
